openedx/adg/lms/applications/helpers.py

```diff
@@ -738,3 +741,46 @@ def create_html_string_for_course_scores_in_admin_review(user_application):
             )
 
     return final_html
+
+
+def update_users_pre_req_courses_status(users_enrolled_in_prereqs, users_with_completed_prereqs, field):
+    """
+    Given the users with active enrollments and users who have completed the courses, the method
+    finds users who have enrolled the pre-reqs but not completed and updates their status in application hub
+
+    Arguments:
+        users_enrolled_in_prereqs (list): list of users
+        users_with_completed_prereqs (list): list of users
+        field (string): application hub field to be updated
+    """
+    from openedx.adg.lms.applications.models import ApplicationHub
+
+    users_with_pre_reqs_in_progress = set(users_enrolled_in_prereqs) - set(users_with_completed_prereqs)
+
+    bulk_update_application_hub_field(field, ApplicationHub.IN_PROGRESS, users_with_pre_reqs_in_progress)
+    bulk_update_application_hub_field(field, ApplicationHub.COMPLETED, users_with_completed_prereqs)
+
+
+def get_users_enrolled_in_prereqs(users_ids, update_bu_prereq=False):
+    """
+    Given the list of user ids, the function returns the users that are either enrolled in program prerequisite
+    courses or business line and common business line prerequisite courses.
+
+    Arguments:
+        users_ids (list): list of user ids
+        update_bu_prereq (bool): boolean flag to get either program pre_reqs or business line pre_reqs
+
+    Return:
+        users_enrolled_in_bu_prereqs (list): list of users
+    """
+    from openedx.adg.lms.applications.models import MultilingualCourseGroup
+
+    prereq_course_groups = (
+        MultilingualCourseGroup.objects.program_prereq_course_groups() if not update_bu_prereq
+        else MultilingualCourseGroup.objects.business_line_and_common_business_line_prereq_course_groups()
+    )
+
+    users_enrolled_in_bu_prereqs = get_users_with_active_enrollments_from_course_groups(
```

Comment:
Rename this to `users_enrolled_in_prereqs` as we are considering both bu and program courses